etrago/cluster/gasclustering.py

```diff
@@ -0,0 +1,397 @@
+            with_time=with_time,
+            custom_strategies=one_port_strategies.get(one_port, {}),
+        )
+        io.import_components_from_dataframe(network_gasgrid_c, new_df, one_port)
+        for attr, df in iteritems(new_pnl):
+            io.import_series_from_dataframe(network_gasgrid_c, df, one_port, attr)
+
+    for c in network.iterate_components(one_port_components):
+        io.import_components_from_dataframe(
+            network_gasgrid_c,
+            c.df.assign(bus=c.df.bus.map(busmap)).dropna(subset=["bus"]),
+            c.name,
+        )
+
+    if with_time:
+        for c in network.iterate_components(one_port_components):
+            for attr, df in iteritems(c.pnl):
+                if not df.empty:
+                    io.import_series_from_dataframe(network_gasgrid_c, df, c.name, attr)
+
+    # Aggregate links
+    new_links = (
+        network.links.assign(
+            bus0=network.links.bus0.map(busmap), bus1=network.links.bus1.map(busmap)
+        )
+        .dropna(subset=["bus0", "bus1"])
+        .loc[lambda df: df.bus0 != df.bus1]
+    )
+
+    new_links["link_id"] = new_links.index
+
+    strategies = {
+        "p_nom": "sum",
+        "length": "mean",
+    }
+    strategies.update(
+        {col: "first" for col in new_links.columns if col not in strategies}
+    )
+
+    gas_carriers = [  # This list should be replace by an automatic selection
+        "CH4",
+        "CH4_to_H2",
+        "H2_feedin",
+        "H2_ind_load",
+        "H2_to_CH4",
+        "H2_to_power",
+        "power_to_H2",
+        "central_gas_CHP",
+        "central_gas_CHP_heat",
+        "industrial_gas_CHP",
+        "rural_gas_boiler",
+        "central_gas_boiler",
+        "OCGT",
+    ]
+
+    gas_links = new_links[new_links["carrier"].isin(gas_carriers)].copy()
+
+    combinations = gas_links.groupby(["bus0", "bus1", "carrier"]).agg(strategies)
+    combinations.reset_index(drop=True, inplace=True)
+
+    combinations["buscombination"] = combinations[["bus0", "bus1"]].apply(
+        lambda x: tuple(sorted([str(x.bus0), str(x.bus1)])), axis=1
+    )
+
+    strategies.update(
+        {col: "first" for col in combinations.columns if col not in strategies}
+    )
+
+    combinations_final = combinations.groupby(["buscombination", "carrier"]).agg(
+        strategies
+    )
+
+    combinations_final.set_index("link_id", inplace=True)
+    combinations_final = combinations_final.drop(columns="buscombination")
+    io.import_components_from_dataframe(network_gasgrid_c, combinations_final, "Link")
+
+    non_gas_links = (
+        new_links[~new_links["carrier"].isin(gas_carriers)]
+        .copy()
+        .drop(columns="link_id")
+    )
+    io.import_components_from_dataframe(network_gasgrid_c, non_gas_links, "Link")
+
+    if with_time:
+        for attr, df in iteritems(network.links_t):
+            if not df.empty:
+                io.import_series_from_dataframe(network_gasgrid_c, df, "Link", attr)

```

Comment:
Time-depended values for links are imported, but there are time series values for some links_id which are not in the links dataframe :
```

WARNING:pypsa.io:Components Index(['1740', '1741', '1742', '1743', '1744', '1745', '1746', '1747', '1748',
       '1749',
       ...
       '2005', '2006', '2007', '2008', '2009', '2010', '2011', '2012', '2013',
       '2014'],
      dtype='object', length=216) for attribute efficiency of Link are not in main components dataframe links
```
I will try to find out why this happens.

Reply:
The warnings seem to be related to #366. It could be solved by changing the order of the clustering and clustering the gas network before clustering the electric grid. Is there anything against changing the execution order like this?

Reply:
I also think it is related to #366. The clustering order must be irrelevant, so if it helps to swap the clusterings, I am not against it.

Reply:
I am working on https://github.com/openego/eTraGo/pull/367, it should be possible to modify the aggragation that in a way that the "in-between" technologies (generalized: carriers that are connected to clustered network structures) are aggregated after the busmaps for gas and AC have been created, indendetly of the clustering order.

Reply:
So how do we want to proceed? Should we wait until #367 is finished and merged into this branch before merging it into dev?

Reply:
For me, it makes sense to merge #367 into this branch first and then check that everything is working together as @ClaraBuettner suggested.

Reply:
This issue will be solved after #377 is merged into dev.